arangod/Agency/AgencyFeature.cpp

```diff
@@ -238,32 +249,36 @@ void AgencyFeature::validateOptions(std::shared_ptr<ProgramOptions> options) {
   // - Statistics: turn off statistics gathering for agency
   // - Action/Script/FoxxQueues/Frontend: Foxx and JavaScript APIs
 
-  std::vector<std::string> disabledFeatures({
-    "MMFilesPersistentIndex", "ArangoSearch", "ArangoSearchAnalyzer",
-    "Statistics", "Action", "Script", "FoxxQueues", "Frontend"});
+  std::vector<std::type_index> disabledFeatures(
+      {std::type_index(typeid(MMFilesPersistentIndexFeature)),
```

Comment:
why don't we have a `constexpr char* ` in each feature with a certain name that contains the features identifier

Reply:
I guess the alternative would be to use a string value populated with some `name()` value, more like we do now. But that's more opaque, as it can't be a static class member, but has to be passed as input to the base constructor. There is also often a mismatch between that name string and the actual feature class name, which can be confusing. I think using the `type_index` lets us enforce more via compiler rather than convention, and that's generally a good thing.

Reply:
Ok, convinced. It also looks like this usage of typeid(type) is resolved at compile time. Then I am happy with it.